### ISSUE TITLE
Thread improvements

### ISSUE THREAD

**Issue by tesshucom** [tesshucom/jpsonic#833]
Related  #824. Rewrite all the old threaded code.  Divide into several stages.
Release timing will probably be split, as it will be a daunting task. For a while, the tasks are a little fluid.

____

This range should be completed in bulk first.

- [x] Create Executors -> #971
- [x] Rewrite thread : LastFMScrobbler
- [x] Rewrite thread : ListenBrainzScrobbler
- [x] Rewrite thread : AudioPlayer
- [x] Rewrite thread : InputStreamReaderThread
- [x] Rewrite thread : TranscodeInputStream
- [x] Rewrite thread : UPnPService
- [x] Rewrite thread : PodcastService
- [x] Rewrite thread : MediaScannerService
- [x] Graceful shutdown -> #829
- [x] v110.0.0: Fix all pools except scan-pool to fixed size. If the size is exceeded, it will fall back to the pooling of the container, so specify the same value for the queue size and corePoolSize.

#### Transparent function changes for each version

<details>
<summary>v110.0.0</summary>
Ultimately, the user should at least be able to adjust the pooling size of the scan. Scans are pooled in v110.0.0, but are served in the same single thread as before. Thread pooling other than that, there is no big problem with fixed adjustment. 

For the short-pool (for general-purpose short processing), the following values are the initial values in v110.0.0. The advantage of use short-task-pool is that they are easier to profile and the task priority can be lower than the common pool.

short-task-pool.corePoolSize=4
short-task-pool.queueCapacity=4
short-task-pool.maxPoolSize=7

Especially when the user is using it alone, a pool of about up to 4 is sufficient for efficient processing of playing music. If you expect multiple users, you don't need a large pool. Fall back to the common pool (jetty: qtp, tomcat: nio) as before.

</details>

v110.1.0 There is nothing special in this version.

_____

From here on, it's a good idea to separate the release versions.

<details>
<summary>v11x.x.x</summary>

##### Dynamic changes in scalability

If anything, the system must always ensure playing and scanning. Other than that, it can be postponed or rejected. We may need derivative features such as adjusting the scalability of the Executor during the scan or disabling certain web page. If you increase the number of parallelization points other than the thread processing that already exists, the number of processes that can be executed at the same time will increase. Adjustment design is required.

##### Subdivision of scanning process

Apart from simple subdivisions to improve processing efficiency, there are processing blocks that must be separated. Genre rewriting and statistics can be mentioned. It needs to be redesigned for scan cancellation or partial scan.

These are after-processes for scanning. Jpsonic has already added some post-processing for scanning. These can also be parallelized in some cases.

##### Other aggressive parallelization

 - For example, the system statistics page used by administrators would be a good sample. A variety of information is listed on the web page, but Info collection can be done in parallel.
 - @Async

</details>

Of the exception handling improvements made in #824, note that `com.tesshu.jpsonic.util.concurrent.ConcurrentUtils#handleCauseUnchecked` is used. Where this happens suggests that the process is a bit complicated. It smells something, especially if it's a Controller. It can be executed as a subthread as Future, or multithreading can be considered.

**Comment by tesshucom** [comment]
Failure is occurring in Dependency check. There are about 10 reports, and it looks like they are being dealt with. Ignore it for a while and proceed.

**Comment by tesshucom** [comment]
Will be taken over by #1373.
